### PR TITLE
React-ify Direction point inputs

### PR DESCRIPTION
**Pull request by bbecquet**
## Description
Migrate the direction panel input fields.
As for the RouteResult part, it can be done separately, mounting the React component when needed in the Panel.
After that we'll be able to migrate the vehicle selector and finally the full direction panel itself.

The `<DirectionInput>` component introduced still uses the same suggestion widget as the previous version, as it works fine and is full vanilla-js. It can wait longer for a migration to React if needed.

The migration simplifies `direction_panel.js` as stuff like focus and style management are now delegated to the input components.

There is some level of complexity though to manage the state of the inputs, as there are still two distinct values to keep: the raw input string as entered by the user and the POI instance which results from the async geocoding.

### CODE REVIEW

File: src/panel/direction/DirectionInput.jsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      <div className={`itinerary_icon itinerary_icon_${pointType}`} />
+      <input
+        ref={this.inputRef}
+        id={`itinerary_input_${pointType}`}
+        className="itinerary_input"
+        type="search"
+        required
+        autoComplete="off"
+        spellCheck="false"
+        placeholder={_(pointType === 'origin' ? 'Start point' : 'End point', 'direction')}
```
Comment (amatissart) [inline src/panel/direction/DirectionInput.jsx]: this syntax will probably not be caught by gettext script:  
`_('Start point')` and `_('End point')`  should appear separately.

File: src/panel/direction/DirectionInput.jsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  componentWillUnmount() {
+    if (this.suggest) {
+      this.suggest.destroy();
+    }
+  }
+
+  onChange = event => {
+    const input = event.target.value;
+    this.props.onChangePoint(input, null);
+    this.suggest.setValue(input);
```
Comment (amatissart) [inline src/panel/direction/DirectionInput.jsx]: As far as I understand, this function is called on each keystroke.
And as a consequence, the suggestions list is flickering between requests.
Maybe due to how `suggest.setValue` resets the suggestions list ?

Actually this looks redundant with the existing mechanism in Autocomplete that creates listeners on "input" events.

File: src/panel/direction/DirectionInput.jsx (hunk 1 above)
Comment (bbecquet) [reply src/panel/direction/DirectionInput.jsx]: Fixed!

File: src/panel/direction/DirectionInput.jsx (hunk 2 above)
Comment (bbecquet) [reply src/panel/direction/DirectionInput.jsx]: Simply removing the line solved the problem, meaning the change handler registered by the autocomplete widget continues to work fine on a React-controlled input.

**Pull request merged**